keybase_backup.py

```diff
@@ -68,56 +57,51 @@ def download(message):
-        folder = chat2folder(message["msg"]["channel"]["name"])
-        attachment_filename = "{}/{}_{}".format(
-            folder,
-            str(message_id),
-            message["msg"]["content"]["attachment"]["object"]["filename"],
-        )
-        download_ask = {}
-        download_ask["method"] = "download"
-        download_ask["params"] = {}
-        download_ask["params"]["options"] = {}
-        download_ask["params"]["options"]["message_id"] = message_id
-        download_ask["params"]["options"]["output"] = attachment_filename
-        download_ask["params"]["options"]["conversation_id"] = conversation_id
-        print("Writing " + attachment_filename)
-        file_exists = os.path.exists(attachment_filename)
-        if not file_exists:
-            kb_call(download_ask)
+        return
+
+    message_id = message["msg"]["id"]
+    conversation_id = message["msg"]["conversation_id"]
+    folder = chat2folder(message["msg"]["channel"]["name"])
+    attachment_filename = "{}/{}_{}".format(
+        folder,
+        str(message_id),
+        message["msg"]["content"]["attachment"]["object"]["filename"],
+    )
+    print("Writing " + attachment_filename)
+    file_exists = os.path.exists(attachment_filename)
+    if not file_exists:
+        kb_call("download", {
+            "conversation_id": conversation_id,
+            "message_id": message_id,
+            "output": attachment_filename,
+        })
 
 
 def get_chat_history(conversation_id, page=None):
     """
     wrapper to get a chat history, base on it's conversation_id
     """
-    chat_history = []
-    chat_history_ask = {}
-    chat_history_ask["method"] = "read"
-    chat_history_ask["params"] = {}
-    chat_history_ask["params"]["options"] = {}
-    chat_history_ask["params"]["options"]["conversation_id"] = conversation_id
-    if page:
-        chat_history_ask["params"]["options"]["pagination"] = {}
-        chat_history_ask["params"]["options"]["pagination"]["next"] = page
-        chat_history_ask["params"]["options"]["pagination"]["num"] = 1000
-    status, chat_history_part = kb_call(chat_history_ask)
-    if status:
-        chat_history = chat_history + chat_history_part["result"]["messages"]
-        last, next_page = get_next(chat_history_part["result"]["pagination"])
-        if not last:
-            chat_history_last_part = get_chat_history(conversation_id, next_page)
-            if chat_history_last_part:
-                chat_history = chat_history + chat_history_last_part
-        return chat_history
-    else:
-        print("Unsuccesfull call")
-        return []
+    options = {
+        "conversation_id": conversation_id,
+    }
+    if page is not None:
+        options = options | {
+            "pagination": {
+                "next": page,
+                "num": 1000,
+            }
+        }
+
+    response = kb_call("read", options)
+
+    messages = response["messages"]
+    last = response["pagination"]["last"]
```

Comment:
need to keep the `try`, as this key does not always exist, or just the `get_next` function

Reply:
My keybase chat history is not big enough to trigger pagination it seems :)

No need to throw/catch a KeyError though, we can either check for the presence of the key, or `.get()` it with a default value as fallback.